zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/tenant/TenantRemoveEntityProcessor.java

```diff
@@ -91,47 +95,90 @@ final var record = command.getValue();
 
   @Override
   public void processDistributedCommand(final TypedRecord<TenantRecord> command) {
-    if (validateEntityAssignment(command)) {
+    if (validateEntityAssignment(command, command.getValue().getTenantId())) {
       stateWriter.appendFollowUpEvent(
           command.getKey(), TenantIntent.ENTITY_REMOVED, command.getValue());
     }
 
     commandDistributionBehavior.acknowledgeCommand(command);
   }
 
-  private boolean validateEntityAssignment(final TypedRecord<TenantRecord> command) {
-    return switch (command.getValue().getEntityType()) {
-      case USER -> validateUserAssignment(command);
+  private boolean validateEntityAssignment(
+      final TypedRecord<TenantRecord> command, final String tenantId) {
+    final var entityType = command.getValue().getEntityType();
+    final var entityId = command.getValue().getEntityId();
+    return switch (entityType) {
+      case USER -> checkUserAssignment(command, tenantId);
+      case GROUP -> checkGroupAssignment(entityId, command, tenantId);
```

Comment:
🔧 We should cover the group case in the engine tests as well.

Reply:
I'l add a test case